pkg/services/ngalert/provisioning/validate.go

```diff
@@ -0,0 +1,15 @@
+package provisioning
+
+type ValidationError struct {
+	msg string
+}
+
+func (e ValidationError) Error() string {
+	return e.msg
+}
+
+func NewValidationError(msg string) ValidationError {
+	return ValidationError{
+		msg: msg,
+	}
+}
```

Comment:
Is there a particular reason why you declared a custom type for error? It makes sense when the type contains more fields than regular error. In this case, I think the simplest approach is to declare an error like 
```
var validationError: error = errors.New("invalid route specification")
```

and then use it like 
```
return fmt.Errorf("%w: route A contains incorrect value", validationError)
```
and upstream you can check whether it is a validation error by

```
if errors.Is(err, validationError) { 
    ...
}
```

Reply:
Only reason is lack of familiarity with how Go errors work. :)

It was purely designed so that we could differentiate a validation error from another type and return the proper status code. I didn't realize you could chain "anonymous" errors like that based only on the string. This is better, thanks!